packages/node/src/edge-functions/edge-handler-template.js

```diff
@@ -5,7 +5,7 @@
 /* global IS_MIDDLEWARE ENTRYPOINT_LABEL */
 
 function buildUrl(requestDetails) {
-  let proto = requestDetails.headers['x-forwarded-proto'];
+  let proto = requestDetails.headers['x-forwarded-proto'].split(',').shift(); // handling multi-protocol like https,http://...
```

Comment:
After doing some Googling, indeed a comma separated list of protocols is valid. I have two concerns:

1. This code doesn't handle whitespace or non-word characters. Using a regex `/\b/` will split on word boundaries so if the header is set to something crazy like `https:` or `https http`, that will also work.

```suggestion
  let proto = requestDetails.headers['x-forwarded-proto'].split(/\b/).shift(); // handling multi-protocol like https,http://...
```

2. I'd love to see a test that demonstrates multiple protocols.

Reply:
Thank you so much, now I'll try to add tests

Reply:
@cb1kenobi Sorry, I'm not sure for which package it is better to add tests: `cli`, `edge` or `node` ?

Reply:
In addition, I can’t successfully execute the `pnpm bootstrap` command, its always fails with error:
```
lerna ERR! EWORKSPACES Bootstrapping with pnpm is not supported. Use pnpm directly to manage dependencies: https://pnpm.io/cli/install
```
![Снимок экрана 2023-02-23 в 01 30 20](https://user-images.githubusercontent.com/2877584/220758058-06b8ea62-ce65-45d4-98e6-0383767942c6.png)

The error looks the same as on my system and on Github Codespaces.

My versions: MacOS `13.2.1`, Node `18.13.0`, pnpm `7.27.1`

Reply:
Usually we just run `pnpm i` instead of bootstrap.

Reply:
@cb1kenobi I need some help, because I do not have sufficient experience in developing Vercel 🙂

Do I understand correctly that I need to use the `createEdgeEventHandler` method from `edge-handler` to create the `Edge Runtime` instance, to which I can send `IncomingMessage` with URL containing a double protocol ?

```ts
import { createEdgeEventHandler } from '../src/edge-functions/edge-handler';
import { IncomingMessage } from 'http';
import { Socket } from 'net';

test('Handle multi-protocol in Edge Functions', async () => {
  const handleEvent = await createEdgeEventHandler(/*...*/)
  const socket = new Socket()
  const req = new IncomingMessage(socket)
  req.url = "https,http://localhost/"
  const result = await handleEvent(req)
  expect(result.statusCode).toEqual(200)
})
```

Reply:
@PonomareVlad that is a great question! The short answer is I don't know. I mentioned this to the team and asked the best way to test this. I didn't get an absolute answer yet. I'll know more tomorrow.

Reply:
I just opened a PR to allow at least unit-level tests of this logic. We can merge my PR first, then update this PR, and add some small tests for handling multiple protocols.
https://github.com/vercel/vercel/pull/9528

Reply:
We had some fun CI issues, but the PR that makes this more testable landed. https://github.com/vercel/vercel/pull/9528

Now you can add a test for this behavior in `packages/node/test/unit/edge-functions/edge-handler-template.test.ts`.

Reply:
Awesome, thank you ! I will try do this tomorrow 🙂